rllib/algorithms/algorithm.py

```diff
@@ -131,11 +131,42 @@
 from ray.util.timer import _Timer
 from ray.tune.registry import get_trainable_cls
 
-
 try:
     from ray.rllib.extensions import AlgorithmBase
 except ImportError:
 
+    def _get_learner_bundle_counts(total_workers, num_workers_per_bundle):
```

Comment:
This helper method lets us default to putting all learner workers onto one node (usually the head node).
But it provides the additional flexibility to spread workers over the cluster (num_workers_per_bundle=1).
Additionally, if we have, say, 2 GPU nodes with four GPUs each, one could set total_workers=8 and num_workers_per_bundle=4 to split the learner workers evenly without accidentally putting them onto smaller nodes meant for rollout workers.

We are obviously still running into the issue of lack of expressiveness when reserving resources with placement groups, but by providing this setting, we can probably cover a lot more use cases where large GPU nodes are used for training and smaller GPU nodes for inference.